backend/main.py

```
@@ -0,0 +1,48 @@
+
+    try:
+        download_image()
+    except Exception as e:
+        os.remove(temp_filename)
+        raise HTTPException(status_code=400, detail=str(e))
+
+    # Verify that the downloaded file is a valid image
+    try:
+        with Image.open(temp_filename) as img:
+            img.verify()
+    except Exception:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)